scripts/setup

```diff
@@ -0,0 +1,63 @@
+#!/bin/bash
```

Comment:
Don't we want scripts to have the .sh extension?

Reply:
It wasn't a part of our original proposal, but happy to modify it

Reply:
... wanted to tag @mflinn-broad on this for the discussion purpose of it. I know he is working on various devops repos on this, too. for context, I made the `.sh` adds locally and didn't want to surprise him in case he didn't have suffixes on his scripts.

Reply:
Mine don't have the .sh extension currently. As a general convention on devops we don't include .sh. But it doesn't really matter to me one way or the other. If folks want .sh I'm fine with that. I think it matters more that we just pick one and stick to it more so than which we actually pick.

Reply:
my original thought here was ...
it doesn't matter if it's python or bash or ??? - the command was the same and its only requirement is that it should be executable. 

adding the extensions means we will be following `setup.sh` and `setup.py` and `setup.rs` and ... look good?

Reply:
That was also my original thought

Reply:
@cahrens - It felt unusual to me at first, too - and totally happy to discuss more if you feel strongly about it. You have a valid point I am definitely not attached to it. 

And to support your point and concern - let's say we say, "write these entry points this way (`setup` / `build` / `run`)." - there is nothing to preventing anyone from encountering the situation where people feel strongly about adding the file suffixes onto it. My then counterpoint to this would be to support (the potential of) having more templatize-ation (not sure if that's a word, but going to roll with it) around our build processes. That would be "the selling point" - if you will. Ultimately, we can afford anything to be parameterized, but then I hit myself with "convention over configuration"-conversation. :sigh: - and yes, my brain is constantly like this.  😬 

Did we follow all of that? - it's totally easy to add back on or take off - it'll be harder after we release this into the wild - so thank you for entertaining the conversation with me - 😇